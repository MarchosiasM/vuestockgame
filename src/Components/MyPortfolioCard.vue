<template>
  <div class="my-portfolio-card col-md-6">
    <p>Name: {{ name }} || Count: {{ count }} || Price: {{ price }} </p>
    <div class="form-group">
      <input v-model="sellCount" type="text">
      <button 
        class="btn btn-primary"
        @click="handleSubmit"
      >Sell</button>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      sellCount: 0
    };
  },
  props: {
    name: String,
    price: Number,
    count: Number
  },
  methods: {
    handleSubmit() {
      const { name, count, price } = this;
      this.$store.dispatch('sellStocks', {
        name,
        count,
        price
      });
    }
  }
};
</script>
