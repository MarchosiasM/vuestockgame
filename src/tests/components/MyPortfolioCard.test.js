import { shallowMount } from '@vue/test-utils';
import MyPortfolioCard from '../../Components/MyPortfolioCard.vue';
import storeObj from './store/store';

const { store: impStore, localVue, actions } = storeObj;

let store;

describe('MyPortfolioCard', () => {
  beforeEach(() => {
    store = Object.assign(impStore);
  });
  it('renders correctly', () => {
    const wrapper = shallowMount(MyPortfolioCard, { localVue, store });
    wrapper.setProps({
      name: 'Pied Piper',
      price: 26,
      count: 30,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('displays correct information passed down as props', () => {
    const wrapper = shallowMount(MyPortfolioCard, { localVue, store });
    wrapper.setProps({
      name: 'Pied Piper',
      price: 26,
      count: 30,
    });
    expect(wrapper.html()).toContain('Pied Piper');
    expect(wrapper.html()).toContain(26);
  });
  it('takes a numerical input and sells stock when you submit', () => {
    const wrapper = shallowMount(MyPortfolioCard, { localVue, store });
    wrapper.setProps({
      name: 'Pied Piper',
      price: 26,
      count: 30,
    });
    const input = wrapper.find('input');
    input.element.value = 100;
    const button = wrapper.find('button');
    button.trigger('click');
    expect(actions.sellStocks).toHaveBeenCalled();
  });
});
