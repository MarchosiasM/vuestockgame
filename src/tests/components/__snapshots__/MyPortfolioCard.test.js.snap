// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPortfolioCard renders correctly 1`] = `
<div
  class="my-portfolio-card col-md-6"
>
  <p>
    Name: Pied Piper || Count: 30 || Price: 26 
  </p>
   
  <div
    class="form-group"
  >
    <input
      type="text"
    />
     
    <button
      class="btn btn-primary"
    >
      Sell
    </button>
  </div>
</div>
`;
